### DISCUSSION TITLE
Lesson 7: Incompatible Solidity Version with LinkToken.sol

### DISCUSSION THREAD

**Discussion by sephirothg** [smartcontractkit/full-blockchain-solidity-course-py#1614] [Q&A] [answered]
Computer: M1 Macbook
Brownie Version: v1.19.0

I'm having the same issue with the one in https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/472

The LinkToken.sol given in the tutorial directory is in `pragma solidity ^0.4.11;`, which brownie tells me the following error:
`CompilerError: solc returned the following errors:`

`contracts/test/LinkToken.sol:2:1: ParserError: Source file requires different compiler version (current compiler is 0.6.6+commit.6c089d02.Darwin.appleclang - note that nightly builds are considered to be strictly less than the released version
pragma solidity ^0.4.11;`
`^----------------------^`

Therefore, I have attempted to use the v0.6 version of LinkToken provided here `[here](https://github.com/smartcontractkit/LinkToken/blob/master/contracts/v0.6/LinkToken.sol)`
and changing my imports in `LinkToken.sol` to 
`import "@linktoken/contracts/v0.6/token/LinkERC20.sol";`
`import "@linktoken/contracts/v0.6/ERC677.sol";`
`import "@linktoken/contracts/v0.6/ITypeAndVersion.sol";`

and added additional dependencies in `brownie-config.yaml` to:
`dependencies:`
`  - smartcontractkit/chainlink-brownie-contracts@1.1.1`
`  - OpenZeppelin/openzeppelin-contracts@3.4.0`
`  - smartcontractkit/linktoken@deployed`
`    remappings:`
`      - '@chainlink=smartcontractkit/chainlink-brownie-contracts@1.1.1'`
`      - '@openzeppelin=OpenZeppelin/openzeppelin-contracts@3.4.0'`
`      - '@linktoken=smartcontractkit/LinkToken'`

Still, brownie cannot compile and given the following errors:

`CompilerError: solc returned the following errors:`

`contracts/test/LinkToken.sol:4:1: ParserError: Source "smartcontractkit/LinkToken/contracts/v0.6/token/LinkERC20.sol" not found: File not found.`
`import "@linktoken/contracts/v0.6/token/LinkERC20.sol";`
^-----------------------------------------------------^

`contracts/test/LinkToken.sol:5:1: ParserError: Source "smartcontractkit/LinkToken/contracts/v0.6/ERC677.sol" not found: File not found.`
`import "@linktoken/contracts/v0.6/ERC677.sol";`
^--------------------------------------------^

`contracts/test/LinkToken.sol:6:1: ParserError: Source "smartcontractkit/LinkToken/contracts/v0.6/ITypeAndVersion.sol" not found: File not found.`
`import "@linktoken/contracts/v0.6/ITypeAndVersion.sol";`
^-----------------------------------------------------^

Please help as I've been stuck here for a few days and no helpful solution has came up across the web.

**Comment by cromewar** [comment, marked as answer]
Hello @sephirothg 
The reason is if you change those files you will have problems with version mismatch as LinkToken just exists on 0.4 but no on 0.6 as the error shown above, follow this instructions:

1. Be sure you solidity versions are the exact same as on the repo (clone it and try to run it to see if it works separately).
2. go to your home folder and delete the `.brownie` folder, and try again.
3.  change the vscode extension to `hardhat-solidity`

Does any of this suggestions work?

**Reply by sephirothg** [reply]
1. I have cloned it and ran it separately and it works ok.
2. I deleted the .brownie folder in my home folder, and it compiled ok this time.
For anyone on a mac who doesn't know (maybe just me) how to reveal folders with ".", press "command" + "shift" + ".", and hidden folders like ".brownie" will be revealed.

Many thanks for your help, my problem is solved.
